General docs update

paddyduke opened the pull request:
This is a general tidy-up before we add some more component docs.

### Note
This is not intended to tidy up the underlying structure of the docs or properly sort out any of its CSS. It's just focused on improving the presentation and the guidance for now.

### Includes:
- **combines** the useful bits of the 'Naming conventions' and 'Creating components' sections together under the latter title
- **removes** the 'General visual rules' section altogether until we have more useful stuff to put in it
- increasing the font size throughout the docs to make them easier to
read
- adjusting spacing around headers and code blocks to make the pages
more spacious and scannable
- improves the link styles
- reducing the size of the header
- some tidying of the menu and adding hover styles
- tidying of the table of contents
- fixing a mistake in a design token screenshot
- fixing a typo in the accessibility page
- improving the appearance of the table of contents on each page
- reworking our CSS code guidelines to better reflect our current rules

summers1981FA commented on assets/scss/docs/_docs.scss:
```diff
@@ -204,23 +206,26 @@ img {
 .masthead-nav {
-  margin-top: 1rem;
+  margin-top: 8px;
   font-size: 1rem;
```
Small thing, alphabetical order wrong here

summers1981FA commented on assets/scss/docs/_docs.scss:
```diff
@@ -253,7 +258,7 @@ img {
-  @include font-weight(normal);
+  font-weight: normal;
   @include margin-bottom(default);
   color: #fff;
```
small thing, alphabetical order wrong here.

summers1981FA commented on assets/scss/docs/_docs.scss:
```diff
@@ -262,7 +267,7 @@ img {
 .Hero-body {
   @include font-size(large);
-  @include line-height(x-tight);
+  line-height: 1.2;
```
no px value here but one of the others above does.

paddyduke commented on assets/scss/docs/components/_table.scss:
```diff
@@ -334,3 +334,13 @@ td .Table--borderless {
+ */
+
+.Table-cell--fitToContents {
+  width: 1%;
```
I didn't write this (just copied it over) but I believe it's to force the cell to shrink.

paddyduke commented on assets/scss/docs/_docs.scss:
```diff
@@ -33,6 +33,12 @@
+}
+
 $masthead-margin: 1.5rem;
 
```
This branch is not really concerned with trying to fix any of the docs CSS. For speed, I'm just working around what's already there and tweaking the bits that needed changes.

I think we can come back separately once we've got some more good content and fix up this aspect.

summers1981FA replied:
np, happy to leave for now

paddyduke commented:
@summers1981FA I've fixed a few of the sorting issues and removed the pixel line-height, but I'm content to leave the rest of the old CSS as it is for now. The docs styles will need a concerted effort at some point to clean up properly.

paddyduke commented:
@summers1981FA As for the table of contents, I think it's basically essential for a lot of these pages, and unless we shift the layout around a bunch, there's not much else we can do with it for now. FWIW I don't think it's a problem that it takes up space. It's a pretty important piece of UI.

summers1981FA commented:
Because its a constant WIP and there i nothing significant broken or removed I am happy to except PR.

More of a concerted effort required on look and feel in the future but not worth holding us up IMO.
